packages/messageport/tizen/src/messageport_tizen_plugin.cc

```diff
@@ -0,0 +1,236 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "messageport_tizen_plugin.h"
+
+#include <flutter/event_channel.h>
+#include <flutter/event_sink.h>
+#include <flutter/event_stream_handler_functions.h>
+#include <flutter/method_channel.h>
+#include <flutter/plugin_registrar.h>
+#include <flutter/standard_method_codec.h>
+
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "log.h"
+#include "messageport.h"
+
+namespace {
+
+class MessageportTizenPlugin : public flutter::Plugin {
+ public:
+  static void RegisterWithRegistrar(flutter::PluginRegistrar *registrar) {
+    auto channel =
+        std::make_unique<flutter::MethodChannel<flutter::EncodableValue>>(
+            registrar->messenger(), "tizen/messageport",
+            &flutter::StandardMethodCodec::GetInstance());
+
+    auto plugin = std::make_unique<MessageportTizenPlugin>(registrar);
+
+    channel->SetMethodCallHandler(
+        [plugin_pointer = plugin.get()](const auto &call, auto result) {
+          plugin_pointer->HandleMethodCall(call, std::move(result));
+        });
+
+    registrar->AddPlugin(std::move(plugin));
+  }
+
+  MessageportTizenPlugin(flutter::PluginRegistrar *pluginRegistrar)
+      : pluginRegistrar_(pluginRegistrar) {}
+
+  virtual ~MessageportTizenPlugin() {}
+
+ private:
+  template <typename T>
+  bool GetValueFromArgs(const flutter::EncodableValue *args, const char *key,
+                        T &out) {
+    if (std::holds_alternative<flutter::EncodableMap>(*args)) {
+      flutter::EncodableMap map = std::get<flutter::EncodableMap>(*args);
+      flutter::EncodableValue value = map[flutter::EncodableValue(key)];
```

Comment:
Unintended insertion occurs If the map doesn't have a key.
Please use a find like line 68